substrafl_examples/go_further/run_mnist_cyclic.py

```diff
@@ -0,0 +1,852 @@
+                Only take into account in ``remote`` mode.
+            round_idx (Optional[int], optional): Current round index. The initialization round is zero by default,
+                but you are free to change it in the ``build_compute_plan`` method. Defaults to 0.
+            additional_orgs_permissions (Optional[set], optional): additional organization ids that could
+                have access to the outputs the task. In our case, this corresponds to the organization
+                containing test data nodes, in order to provide access to the model and to allow to
+                use it on the test data.
+        """
+        first_train_data_node = train_data_nodes[0]
+
+        # The algo.initialize method is an empty method useful to load all python object to the platform.
+        self._cyclic_local_state = first_train_data_node.init_states(
+            operation=self.algo.initialize(
+                _algo_name=f"Initializing with {self.algo.__class__.__name__}",
+            ),
+            round_idx=round_idx,
+            authorized_ids=set([first_train_data_node.organization_id]) | additional_orgs_permissions,
+            clean_models=clean_models,
+        )
+
+    def perform_round(
+        self,
+        *,
+        train_data_nodes: List[TrainDataNode],
+        aggregation_node: Optional[AggregationNode],
+        round_idx: int,
+        clean_models: bool,
+        additional_orgs_permissions: Optional[set] = None,
+    ):
+        """This method will be call at each round to perform a series of task. For the cyclic
+        strategy we want to design, a round is a full cycle over the different train data
+        nodes.
+        We link the output of a computed task directly to the next one.
+
+        Args:
+            train_data_nodes (List[TrainDataNode]): Train data nodes representing the different
+                organizations containing data we want to train on.
+            aggregation_node (List[AggregationNode]): In the case of the Cyclic Strategy, there is no
+                aggregation tasks so no need for AggregationNode.
+            clean_models (bool): Boolean to indicate if we want to keep intermediate shared state.
+                Only take into account in ``remote`` mode.
+            round_idx (Optional[int], optional): Current round index.
+            additional_orgs_permissions (Optional[set], optional): additional organization ids that could
+                have access to the outputs the task. In our case, this will correspond to the organization
+                containing test data nodes, in order to provide access to the model and to allow to
+                use it on the test data.
+        """
+        for i, node in enumerate(train_data_nodes):
+            # We get the next train_data_node in order to add the organization of the node
+            # to the authorized_ids
+            next_train_data_node = train_data_nodes[(i + 1) % len(train_data_nodes)]
```

Comment:
there might be a smarter way of doing that using `itertools.cycle`, but it's not very important

Reply:
https://docs.python.org/3.9/library/itertools.html#itertools.cycle

Reply:
Mmmmh it's not that straightforward :/